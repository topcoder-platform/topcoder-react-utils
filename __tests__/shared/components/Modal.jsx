import Modal, { BaseModal } from 'components/Modal';
import React from 'react';
import ReactDom from 'react-dom';
import {
  findInDomByClass,
  renderDom,
  simulate,
  snapshot,
} from 'utils/jest';

beforeAll(() => {
  /* Modal uses ReactJS portals to ensure proper rendering. react-test-renderer,
   * used by utils/jest under the hood, does not support it properly, thus this
   * simple mock for the createPortal(..) function. */
  ReactDom.createPortal = jest.fn(element => (
    <div className="MOCK_PORTAL">
      {element}
    </div>
  ));
});

test('Snapshot match', () => {
  snapshot(<Modal>CONTENT</Modal>);
});

test('onCancel', () => {
  const onCancel = jest.fn();
  const dom = renderDom((
    <BaseModal
      onCancel={onCancel}
      theme={{ overlay: 'overlay' }}
    />
  ));
  const overlay = findInDomByClass(dom, 'overlay');
  simulate.click(overlay);
  expect(onCancel).toHaveBeenCalled();
});

test('onWheel', () => {
  const dom = renderDom((
    <BaseModal
      theme={{ container: 'container' }}
    />
  ));
  const container = findInDomByClass(dom, 'container');
  const stopPropagation = jest.fn();
  simulate.wheel(container, { stopPropagation });
  expect(stopPropagation).toHaveBeenCalled();
});
