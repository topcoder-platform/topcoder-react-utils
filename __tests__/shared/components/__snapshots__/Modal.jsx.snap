// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot match 1`] = `
<div
  className="MOCK_PORTAL"
>
  <div
    className="style__container___1SoeQ"
    onWheel={[Function]}
  >
    CONTENT
  </div>
  <button
    className="style__overlay___X12tn"
    onClick={[Function]}
    type="button"
  />
</div>
`;
